framework/log/FileTarget.php

```diff
@@ -112,6 +112,13 @@ public function export()
         }
 
         $text = implode("\n", array_map([$this, 'formatMessage'], $this->messages)) . "\n";
+
+        $trimmedText = trim($text);
+
+        if (empty($trimmedText)) {
```

Comment:
Can we move this whole check to the first line? To not create directories if there is nothing to export anyway.
Also - why not use `$trimmedText` later on when actually saving to the file?